pkg/kv/kvserver/replica_raft.go

```diff
@@ -1092,9 +1107,25 @@ func (r *Replica) refreshProposalsLocked(
 		log.Fatalf(ctx, "refreshAtDelta specified for reason %s != reasonTicks", reason)
 	}
 
+	var maxInflightDurationReq *roachpb.BatchRequest
+	var maxInflightDuration time.Duration
+	var slowProposalCount int64
 	var reproposals pendingCmdSlice
 	for _, p := range r.mu.proposals {
-		if p.command.MaxLeaseIndex == 0 {
+		slowReplicationThreshold, ok := r.slowReplicationThreshold(p.Request)
+		inflightDuration := r.store.cfg.RaftTickInterval * time.Duration(r.mu.ticks-p.createdAtTicks)
```

Comment:
Do we really want to rely on ticks here? Since ticks are driven by the scheduler, they can drift significantly from real time if the scheduler is slow (see #66686).

Reply:
I think it's worth making a note and revisiting later. If the scheduler is overwhelmed it could cause ranges to essentially be unavailable but the breakers would be very slow to trip. But no need to block this PR on that.